### ISSUE TITLE
New MC Fit to Kaon Data from JE

### ISSUE THREAD

**Issue by jethier** [JeffersonLab/IMC4FF#5]
I present the first MC fit to SIA kaon data. The details of my fit are the following:

- Used 'iterative' evolution with JAM alpha_s (matching='I') and charm mass m_c = 1.275 and bottom mass m_b = 4.18 GeV (MSbar scheme listed by PDG)

- The number of flavors in the FF evolution increased when Q2 reached heavy quark mass values (i.e. Nf=4 when Q2>=m_c^2), but the number of flavors in the SIA cross section was determined by twice the quark mass (i.e. Nf=4 when Q2>=(2.*m_c)^2)

-  I located and corrected the bug that Nobuo found in the 'iterative' part of the evolution code

- The theory was calculated at the average < z > values given directly from experimental papers, or via simple calculation of < z > from average < p > or < x_p > values given by the experiment. The exception to this is DELPHI, in which I take the center of the momentum bin as < p > for low-p bins (the theory values differ by 0.05% compared to an average of the theory across the low-p bins) and average the cross section across high-p bins.

- 100 fits were performed with starting parameters sampled randomly within a close vicinity of HKNS parameters. Parameterization is the same as HKNS and all parameters were free (gluon alpha parameter is no longer fixed)

The following plots are z*FF:

![up](https://cloud.githubusercontent.com/assets/12139833/13365758/ea3677ae-dca4-11e5-9111-05e3acc25e95.jpg)

![sbar](https://cloud.githubusercontent.com/assets/12139833/13365787/175d3fd8-dca5-11e5-9cf2-70081cd00e7d.jpg)

![ubar](https://cloud.githubusercontent.com/assets/12139833/13365800/2e9f52f8-dca5-11e5-8ac3-4b3fbe74b6e9.jpg)

![charm](https://cloud.githubusercontent.com/assets/12139833/13366018/9183f170-dca6-11e5-9bf0-f988238b09e6.jpg)

![bottom](https://cloud.githubusercontent.com/assets/12139833/13366022/945ebb46-dca6-11e5-8df1-232b208033be.jpg)

![glue](https://cloud.githubusercontent.com/assets/12139833/13366090/164741a0-dca7-11e5-97d4-a08ea987f6e3.jpg)

**Comment by nobuosato** [comment]
Interesting... Jake may I suggest showing one single fit starting from HKNS parameters and also a summary of the chi2

**Comment by nobuosato** [comment]
Also the b distribution is strange.... I would expect this to identical to HKNS... Let's see how does your single fits look like

**Comment by jethier** [comment]
Results from a single fit of SIA kaon data with everything the same as above except the following differences:

- Gluon alpha parameter is fixed as in HKNS to 5.0

- No data resampling or partitioning of data into training/validation sets (MMC)

## chi2
| exp         | npts | chi2(HKNS) NLO | chi2(IMC) NLO |
| :--:        | :--: | :--:          | :--:         | :--:           | :--:          |
| TASSO       | 18   | 25.0           | 18.1          |
| TPC         | 17  | 15.2         | 14.0          |
| HRS         | 3    | 0.4            | 0.7           |
| TOPAZ       | 3    | 0.8            | 0.6           |
| SLD         | 29   | 12.3           | 12.1          |
| SLD(uds)    | 29   | 57.2           | 51.4          |
| SLD(c)      | 29   | 32.4           | 32.2          |
| SLD(b)      | 28   | 88.7           | 88.0          |
| ALEPH       | 18   | 12.8           | 6.5          |
| OPAL        | 10   | 11.5           | 11.5          |
| DELPHI      | 27   | 15.2           | 13.2          |
| DELPHI(uds) | 17   | 22.1           | 17.6          |
| DELPHI(b)   | 17   | 11.7           | 12.2          |
| TOTAL       | 245  | 305.3          | 278.1         |

The following are the plots: 

![up](https://cloud.githubusercontent.com/assets/12139833/13375813/0db0a5a0-dd78-11e5-96b5-9f0386218db9.png)

![sbar](https://cloud.githubusercontent.com/assets/12139833/13375815/16461358-dd78-11e5-8748-ca3f4cb70e48.png)

![ubar](https://cloud.githubusercontent.com/assets/12139833/13375816/1bbed5f4-dd78-11e5-9108-54e86de9dc06.png)

![charm](https://cloud.githubusercontent.com/assets/12139833/13375818/227b73d4-dd78-11e5-8be9-becdbb74fd95.png)

![bottom](https://cloud.githubusercontent.com/assets/12139833/13375819/2695a25a-dd78-11e5-8a22-4bd6adad1554.png)

![glue](https://cloud.githubusercontent.com/assets/12139833/13375820/3012aa26-dd78-11e5-8f9b-a5e31431a7d3.png)

**Comment by nobuosato** [comment]
Thanks, Qualitatively, The shifts in FF are similar in your fit and my fit. However, In my fit I find that c and b do not change at all with respect to HKNS, While in your case the b has some significant shift,, even after data re-sampling. 

There are few points that I can think off

- Treatment of HQ. You mentioned that HQ are activated in at the xsection when Q=2m_HQ. I think you should do it at Q=m_HQ in order to compare with HKNS. because this is what they do. If the changes in b and c are due to this then is an interesting point to discuss in the paper. Do you know what DSS do?. 

- Your chi2s are a bit smaller than what I got. Some cross checks.
   1) are you using 100 points for the gaussian quadrature in the inverse mellin. This turns out to be crucial for me 
   2) Are you including normalization uncertainties added in quadrature? if so you should take them off as HKNS do not use those errors. Only what is quoted in the tables as errors. If a given experiment has stat and syst then they add this in quadrature.  

Also in your comment above, could you include the values of the parameters?

**Comment by nobuosato** [comment]
Jake, did you resample the data in your 100 MC fits?

**Comment by jethier** [comment]
Yes, I did!

**Comment by jethier** [comment]
> Treatment of HQ. You mentioned that HQ are activated in at the xsection when Q=2m_HQ. I think you should do it at Q=m_HQ in order to compare with HKNS. because this is what they do. If the changes in b and c are due to this then is an interesting point to discuss in the paper. Do you know what DSS do?.

According to HKNS paper, the number of active flavors in the cross section increases at Q=2*m_HQ as stated in section III-A-1 (Pion), so I was following HKNS. However, most of the experimental data are at a Q2 where this treatment doesn't matter. DSS does not mention doing anything of this sort.

> 1) are you using 100 points for the gaussian quadrature in the inverse mellin. This turns out to be crucial for me 
2) Are you including normalization uncertainties added in quadrature? if so you should take them off as HKNS do not use those errors. Only what is quoted in the tables as errors. If a given experiment has stat and syst then they add this in quadrature.

1. Yes, I am using 100 point gaussian.
2. I treat all uncertainties as they are given in the table and how the chi2 is defined in JAM, so I believe I am including correlated normalization uncertainties in the definition of the chi2. I can re-fit without these correlated uncertainties and see what I get.

> Also in your comment above, could you include the values of the parameters?

## NLO parameters
| param | M(HKNS)         | a(HKNS)        | b(HKNS)        | M(IMC) | a(IMC)  | b(IMC)  |
| :--:  | :--:            | :--:           | :--:           | :--:   | :--:    | :--:    |
| u     | 0.0740  | −0.630  | 1.310   | 0.0486 | -0.5286 | 0.8226  |
| sb    | 0.0878  | 2.000   | 2.800   | 0.0419 | 9.5400 | 5.8634  |
| ub    | 0.0255  | 1.012   | 8.000   | 0.0438 | 0.5280  | 4.9482  |
| c     | 0.0583  | 0.527   | 5.866   | 0.0590 | 0.5891  | 5.9791  |
| b     | 0.0522  | 0.247   | 11.212  | 0.0514 | 0.3182  | 11.5486 |
| g     | 0.0705  | 5.0  | 0.810   | 0.0700 | 5.0000  | 1.2008  |

**Comment by jethier** [comment]
So I found out why there is a difference between the charm and bottom distributions. I was using the HKNS fortran code for the HKNS curves in the above plots, which has a discrepancy with the parameterization at the input scale because of interpolation in the fortran code. If I use the parameters above, these are the curves for the charm and bottom:

![charm](https://cloud.githubusercontent.com/assets/12139833/13412040/9bf2016e-df0e-11e5-87fd-df856bd9b623.png)

![bottom](https://cloud.githubusercontent.com/assets/12139833/13412044/a2199db8-df0e-11e5-9662-5d0c3642bb01.png)

As you can see, there is very little difference between the curves. I suspect the bottom fragmentation distribution from HKNS will be consistent with the MC fits in the first post if I use the HKNS parameters from the paper rather than the fortran code.

**Comment by nobuosato** [comment]
Good!!!.... did you made the fit using the regular chi2 definition?

**Comment by jethier** [comment]
I did do a fit with the regular chi2 definition and I see no noticeable difference from the plots above. As for sampling the initial set of parameters, I have pmin and pmax set within a range very close to HKNS parameters (e.g. for the M parameter for the up distribution I set pmin=0.07 and pmax=0.08 where HKNS is 0.0740)

**Comment by nobuosato** [comment]
I would like to see the chi2 values with the regular chi2 and compare them with mine. Do we get the same? We should get identical results.
